sdk/ai/Azure.AI.Inference/src/Customized/AIInferenceExtensions.cs

```diff
@@ -44,20 +50,30 @@ private static ChatCompletionsClient CreateChatCompletionsClient(this Connection
         /// <returns></returns>
         public static EmbeddingsClient GetEmbeddingsClient(this ConnectionProvider provider)
         {
-            EmbeddingsClient embeddingsClient = provider.Subclients.GetClient(() => CreateEmbeddingsClient(provider), null);
+            EmbeddingsClientKey embeddingsClientKey = new();
+            EmbeddingsClient embeddingsClient = provider.Subclients.GetClient(embeddingsClientKey, () => CreateEmbeddingsClient(provider));
             return embeddingsClient;
         }
 
         private static EmbeddingsClient CreateEmbeddingsClient(this ConnectionProvider provider)
         {
             ClientConnection connection = provider.GetConnection(typeof(ChatCompletionsClient).FullName!);
+
             if (!connection.TryGetLocatorAsUri(out Uri? uri) || uri is null)
             {
                 throw new InvalidOperationException("Invalid URI.");
             }
-            return connection.Authentication == ClientAuthenticationMethod.Credential
-            ? new EmbeddingsClient(uri, connection.Credential as TokenCredential)
-            : new EmbeddingsClient(uri, new AzureKeyCredential(connection.ApiKeyCredential!));
+
+            return connection.CredentialKind switch
+            {
+                CredentialKind.ApiKeyString => new EmbeddingsClient(uri, new AzureKeyCredential((string)connection.Credential!)),
+                CredentialKind.TokenCredential => new EmbeddingsClient(uri, (TokenCredential)connection.Credential!),
+                _ => throw new InvalidOperationException($"Unsupported credential kind: {connection.CredentialKind}")
+            };
         }
+
+        private record ChatCompletionsClientKey() : IEquatable<object>;
```

Comment:
Could we leave private record ChatCompletionsClientKey(AzureAIInferenceClientOptions? Options = null) : IEquatable<object>; here to use it in futire for the UserAgent?

Reply:
This PR is specifically for the changes in SCM. We can add it future PR.